### PR TITLE
Implement OpenTelemetry tracing

### LINKED ISSUE #171: Incorporate OpenTelemetry for Tracing

### PR DESCRIPTION
**Pull request by ryanmelt**
closes #171 

To test add to compose.yaml:

```
openc3-jaeger:
  image: "${OPENC3_REGISTRY}/jaegertracing/all-in-one:1.38"
  command: ["--memory.max-traces=1000000"]
  ports:
    - "127.0.0.1:16686:16686"
  restart: on-failure
  environment:
    - "COLLECTOR_OTLP_ENABLED=true"
  env_file:
    - ".env"
 ```

Add to .env:

```
OTEL_EXPORTER_OTLP_ENDPOINT=http://openc3-jaeger:4318
OPENC3_OTEL=CMD__TLM__API
```

### CODE REVIEW

File: openc3/lib/openc3/utilities/open_telemetry.rb [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+          end
+          c.use 'OpenTelemetry::Instrumentation::Redis', {
+            # The obfuscation of arguments in the db.statement attribute is enabled by default.
+            # To include the full query, set db_statement to :include.
+            # To obfuscate, set db_statement to :obfuscate.
+            # To omit the attribute, set db_statement to :omit.
+            db_statement: :include,
+          }
+          c.use 'OpenTelemetry::Instrumentation::HttpClient'
+          c.use 'OpenTelemetry::Instrumentation::AwsSdk'
```
Comment (jmthomas) [inline openc3/lib/openc3/utilities/open_telemetry.rb:84]: All these `use` statements are how OpenTelemetry can instrument and interpret various nested calls within Ruby?

**Review by jmthomas — APPROVED** [review]
I look forward to trying it out!

File: openc3/lib/openc3/utilities/open_telemetry.rb [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  def self.inject_context(hash)
+    if @otel_enabled
+      OpenTelemetry.propagation.inject(hash)
+    end
+  end
+
+  def self.with_context(hash)
+    if @otel_enabled
+      extracted_context = OpenTelemetry.propagation.extract(hash)
+      OpenTelemetry::Context.with_current(extracted_context) do
```
[the remark this answers is not in the document]
Comment (ryanmelt) [reply openc3/lib/openc3/utilities/open_telemetry.rb:32]: Yep

File: openc3/lib/openc3/utilities/open_telemetry.rb (hunk 1 above)
Comment (ryanmelt) [reply openc3/lib/openc3/utilities/open_telemetry.rb:84]: The use statements enable built in instrumentors provided by OpenTelemetry.  So they automatically inject spans for calls to the associated libraries.

**Pull request merged**